editions/tw5.com/tiddlers/wikitext/Linking in WikiText.tid

```diff
@@ -80,3 +80,40 @@ You can also use the extended syntax to force an external link:
 
 See the LinkWidget for details of the underlying widget used to implement tiddler links, including macros that can be used to customise its behaviour.
 
+! Generating dynamic links
+
+[[Linking in WikiText]] does not lend itself well to creating dynamic links. The reason is because this WikiText link:
+
+```
+[[link to myself|Linking in WikiText]]
+```
+is shorthand for using the LinkWidget and TextWidget like this:
+
+```
+<$link to="Linking in WikiText">
+  <$text text="link to myself"/>
+</$link>
+```
+Since both the link title ("link to myself") and the link target ("Linking in ~WikiText") are used as widget attributes, no WikiText expansion takes place.
+
+For example, an attempt to use a [[reference|TextReference]] and a [[variable|Variables]] to dynamically generate a link like this:
+
+```
+[[{{!!caption}}|<<currentTiddler>>]]
+```
+is the same as trying to use the LinkWidget and TextWidget like this:
+
+```
+<$link to="<<currentTiddler>>">
```

Comment:
I'm concerned that examples of what doesn't work or not to do might be misinterpreted by people scanning the text, and not reading it all properly.

We did attempt to devise a styling to indicate non-functional examples:

https://tiddlywiki.com/#Concatenating%20text%20and%20variables%20using%20macro%20substitution

It's not wildly successful, and has yet to be made into a macro, but perhaps this is the time to introduce a `<<.bad-example "">>` docs macro

Reply:
Thanks for the feedback and good idea. I created a `<<.bad-example "">>` docs macro as you suggested and I made use of it for the text in my original commit as well as for the text in the tiddler you referenced.